addon-test-support/assertions/expect-deprecation.js

```diff
@@ -0,0 +1,37 @@
+import {
+  getDeprecationsDuringCallback,
+  getDeprecations,
+} from '@ember/test-helpers';
+import checkMatcher from './utils/check-matcher';
+
+export default function expectDeprecation(cb, matcher) {
```

Comment:
consider `matcher` also being an array of deprecation messages, or matchers. This would let us unify with the above deprecation helpers